discovery_engine_core/core/src/stack/filters/article.rs

```diff
@@ -282,6 +293,21 @@ mod tests {
         assert!(result.is_empty());
     }
 
+    #[test]
+    fn test_filter_sources() {
+        let articles: Vec<Article> =
+            serde_json::from_str(include_str!("../../../test-fixtures/articles-valid.json"))
+                .unwrap();
+        assert_eq!(articles.len(), 4);
+
+        // all 4 articles have source domain example.com
+        let filtered1 = SourcesFilter::apply(articles, &["bbc.com".to_string()]);
```

Comment:
:art: we should refrain from using real domains, to make the test still work we could just use a different top level domain
```suggestion
        let filtered1 = SourcesFilter::apply(articles, &["example.org".to_string()]);
```